### PR TITLE
kwalitee: addition of sentence dot check

### PR DESCRIPTION
**Pull request by david-caro**

### CODE REVIEW

**Comment by jirikuncar** [comment]
@david-caro there are another valid ending: `* Fixes foo bar.  (closes #14)` or `* Fixes foo long\n  message.  (addresses #1)`.

**Comment by david-caro** [comment]
Sorry, missed that one. the parenthesis there is something defined? must be on the same line?

File: kwalitee/kwalitee.py [hunk 1]
```diff
@@ -141,17 +142,26 @@ def _check_bullets(lines, **kwargs):
     """
     max_length = kwargs.get("max_length", 72)
     labels = {l for l, _ in kwargs.get("commit_msg_labels", tuple())}
+    signatures = kwargs.get("signatures", tuple())
+
+    def _is_signature(line):
```
Comment (jirikuncar) [inline kwalitee/kwalitee.py]: why do you need it here?

File: kwalitee/kwalitee.py [hunk 2]
```diff
@@ -164,6 +174,14 @@ def _check_bullets(lines, **kwargs):
                     errors.append(("M121", i + j + 3))
                 else:
                     skipped.append(i + j + 1)
+                    if not _is_signature(indented):
```
Comment (jirikuncar) [inline kwalitee/kwalitee.py]: "find a first line before bullet or an empty line"

File: kwalitee/kwalitee.py (hunk 1 above)
Comment (david-caro) [inline kwalitee/kwalitee.py]: To properly find the end of the bullet message, as the signatures don't have a dot they should be treated differently and never be considered part of the dotted message itself.

File: kwalitee/kwalitee.py (hunk 2 above)
Comment (david-caro) [inline kwalitee/kwalitee.py]: Sorry, can you rephrase/elaborate?

Comment (jirikuncar) [inline kwalitee/kwalitee.py]: You don't need to check for signatures in `_check_bullets`.

Comment (jirikuncar) [inline kwalitee/kwalitee.py]: Signatures are separated with an empty line and are not indented to you don't need to check for them here.

Comment (david-caro) [inline kwalitee/kwalitee.py]: good to know

**Comment by david-caro** [comment]
Sorry for the noise, for now I just expect the '()' to not be multiline, added tests for that case too.

**Comment by jirikuncar** [comment]
IMHO we can do it with a regex `\.([ ]\{1,\}\([^)]\))?$`.

**Comment by david-caro** [comment]
that does not work if the comment is like:
```
* bulleted thingie.
  (closes #123)
```

**Comment by jirikuncar** [comment]
True. So we need something like `/^  (.*\.)?([ ]\{1,\}\([^)]\{1,\}\))?$/` for the last line of a paragraph?

**Comment by david-caro** [comment]
at that point in the code, we have a line-by-line loop, so I would have to go reconstructing them. Looks more complicated (also, the regex starts to be complicated too...)

**Comment by jirikuncar** [comment]
But you know which is the last line of the bullet point so you can just check that one.

File: kwalitee/kwalitee.py [hunk 3]
```diff
@@ -142,16 +144,22 @@ def _check_bullets(lines, **kwargs):
     max_length = kwargs.get("max_length", 72)
     labels = {l for l, _ in kwargs.get("commit_msg_labels", tuple())}
 
+    def _strip_ticket_directives(line):
```
Comment (jirikuncar) [inline kwalitee/kwalitee.py]: hmm it looks fine.

**Comment by jirikuncar** [comment]
Cool. I think we can integrate it. Thanks a lot for quick replies!

File: kwalitee/kwalitee.py [hunk 4]
```diff
@@ -142,16 +144,22 @@ def _check_bullets(lines, **kwargs):
...
+        return re.sub(r' \([^)]*\)$', '', line)
```
Comment (jirikuncar) [inline kwalitee/kwalitee.py]: there might me several `* Foo bar. (closes #1) (addresses #2)`

Comment (jirikuncar) [inline kwalitee/kwalitee.py]: `* -> \{1,\}`

**Comment by david-caro** [comment]
Found also another issue, added tests for it too

**Comment by david-caro** [comment]
Is there anything else I should address?

**Comment by david-caro** [comment]
oops... did not refresh the page :see_no_evil:

**Comment by jirikuncar** [comment]
@david-caro if you have something just open new PR.
